tests/test_targets.py

```diff
@@ -295,3 +296,69 @@ def test_s3_upload(mock_client):
     )
 
     assert resp == "my test file"
+
+
+def test_csv_upload():
```

Comment:
Question: do you think it would make sense to split up this test case into three (according to the three `target.data_received(...)` calls made in this test)?

Reply:
Good suggestion. To be honest I prefer test cases covering only one use case per test whenever possible as they are easier to understand and maintain. Let me know if how split them makes sense.

Reply:
Looks good how it is now. 👍🏼